File: lab5/Token.java
package lab5;

public class Token {
    private final Position startPosition, endPosition;
    private final String tagName, lexeme;

    Token(String tagName, Position startPosition, Position endPosition, String lexeme) {
        this.tagName = tagName;
        this.startPosition = startPosition;
        this.endPosition = endPosition;
        this.lexeme = lexeme;
    }

    @Override
    public String toString() {
        return  this.tagName + " " + this.startPosition  + "-" + this.endPosition + ": " + this.lexeme;

    }
}

File: lab7_1/lexer_ts/src/Token.ts
import { Position } from "./Position"

class Token {

    private startPosition: Position;
    private endPosition: Position;
    private lexeme: string
    private tagName: string;

    constructor(tagName: string, startPosition: Position, endPosition: Position, lexeme: string) {
        this.tagName = tagName;
        this.startPosition = startPosition;
        this.endPosition = endPosition;
        this.lexeme = lexeme;
    }

    public getTag(): string {
        return this.tagName;
    }

    public toString(): string {
        return this.tagName  + this.startPosition.toString() + '-' + this.endPosition.toString() + ': ' + this.lexeme;
    }
}


export { Token }